server/schedulers/balance_region.go

```diff
@@ -165,3 +174,29 @@ func (s *balanceRegionScheduler) transferPeer(cluster schedule.Cluster, region *
 	balanceRegionCounter.WithLabelValues("move_peer", fmt.Sprintf("store%d-in", target.GetId())).Inc()
 	return schedule.CreateMovePeerOperator("balance-region", cluster, region, schedule.OpBalance, oldPeer.GetStoreId(), newPeer.GetStoreId(), newPeer.GetId())
 }
+
+// hasPotentialTarget is used to determine whether the specified sourceStore
+// cannot find a matching targetStore in the long term.
+// The main factor for judgment includes StoreState, DistinctScore, and
+// ResourceScore, while excludes factors such as ServerBusy, too many snapshot,
+// which may recover soon.
+func (s *balanceRegionScheduler) hasPotentialTarget(cluster schedule.Cluster, region *core.RegionInfo, source *core.StoreInfo, opInfluence schedule.OpInfluence) bool {
+	filters := []schedule.Filter{
+		schedule.NewExcludedFilter(nil, region.GetStoreIds()),
+		schedule.NewDistinctScoreFilter(cluster.GetLocationLabels(), cluster.GetRegionStores(region), source),
+	}
+
+	for _, store := range cluster.GetStores() {
+		if schedule.FilterTarget(cluster, store, filters) {
+			continue
+		}
+		if !store.IsUp() || store.DownTime() > cluster.GetMaxStoreDownTime() {
```

Comment:
Why not `>=` ?

Reply:
They are not much different. And the `healthFilter` use `>` too.
https://github.com/pingcap/pd/blob/f7f11444a5583249d896072d226db92f5bd5cc2d/server/schedule/filters.go#L140